ipynb/AG_stool_temporal dynamics.ipynb

```diff
@@ -0,0 +1,664 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import skbio\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "dm = skbio.DistanceMatrix.read('ag-cleaned-multiple-timepoints_even1250_unweighted.txt')\n",
+    "w_dm = skbio.DistanceMatrix.read('ag-cleaned-multiple-timepoints_even1250_weighted.txt')\n",
+    "deltas = pd.read_csv('ag-cleaned-multiple-timepoints-deltas.txt', sep='\\t', dtype=str)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "absdeltas = deltas"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "deltas['DeltaInDays'] = pd.to_numeric(deltas['DeltaInDays'])\n",
+    "absdeltas['DeltaInDays'] = deltas['DeltaInDays'].abs()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "565\n",
+      "1553\n",
+      "1553\n"
+     ]
+    }
+   ],
+   "source": [
+    "unique_ids = []\n",
+    "unique_sampleA = []\n",
+    "unique_sampleB = []\n",
+    "\n",
+    "for i in deltas['HOST_SUBJECT_ID']:\n",
+    "    if i not in unique_ids:\n",
+    "        unique_ids.append(i)\n",
+    "        \n",
+    "for i in deltas['SampleID-A']:\n",
+    "    if i not in unique_sampleA:\n",
+    "        unique_sampleA.append(i)\n",
+    "        \n",
+    "for i in deltas['SampleID-B']:\n",
+    "    if i not in unique_sampleB:\n",
+    "        unique_sampleB.append(i)\n",
+    "\n",
+    "print(len(unique_ids))\n",
+    "print(len(unique_sampleA))\n",
+    "print(len(unique_sampleB))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "distance = []\n",
+    "for dl in range(len(absdeltas)):\n",
+    "    distance.append(dm[absdeltas['SampleID-A'][dl], absdeltas['SampleID-B'][dl]])\n",
+    "\n",
+    "absdeltas['Distance'] = distance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "delmonths = []\n",
+    "for dl in range(len(absdeltas)):\n",
```

Comment:
Can you provide an explanation in the document about what is it that you are doing here?